commands/logHw.ts

```diff
@@ -0,0 +1,100 @@
+import { CommandInteraction, TextChannel } from 'discord.js';
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { addLogHwOptions } from '../common/commandHelper';
+import { getClass } from '../api/classApi';
+import { getMessageChannel } from '../api/messageChannelApi';
+import { getHomeworks } from '../api/homeworkApi';
+import { DateValidator } from '../common/logbook-date';
+import { HomeworkLogBook } from '../common/logbook-homework';
+import { ICommand } from '../types/command';
+
+const slashCommandBuilder = new SlashCommandBuilder();
+slashCommandBuilder.setName('loghw').setDescription('Log a club in the message channel.').setDefaultPermission(false);
+addLogHwOptions(slashCommandBuilder);
+
+export const logHw: ICommand = {
+    config: slashCommandBuilder,
+    execute: async (interaction: CommandInteraction) => {
+        await interaction.deferReply();
+        const { channel, client, options } = interaction;
+        let startDay = options.getString('start_date');
+        const startTime = options.getString('start_time');
+        let endDay = options.getString('end_date');
+        const endTime = options.getString('end_time');
+        const classCode = options.getString('class_code');
+        const desc = options.getString('description') || '';
+        const hwDesc = options.getString('hw_description') || 'Assignment [number]';
+        const shouldNotAllowMultipleEntries = options.getBoolean('no_multiples');
+
+        if (classCode.length >= 7) {
+            await interaction.followUp('class_code should have 6 characters.');
+            return;
+        }
+
+        if (
+            !DateValidator.isValidDate(endDay) ||
+            !DateValidator.isValidDate(startDay) ||
+            !DateValidator.isValidTime(startTime) ||
+            !DateValidator.isValidTime(endTime)
+        ) {
+            await interaction.followUp('Please insert the correct format for dates and time (YYYY/MM/DD HH:MM)');
+            return;
+        }
+        DateValidator.adaptFormatOfDays(startTime, startDay, endTime, endDay);
```

Comment:
Why do we need to call this func twice, once without and once with saving the result?

Reply:
we don't. 😮 it was previously there, but i removed. nice find!